lib/realtime/channels.ex

```diff
@@ -54,4 +54,19 @@ defmodule Realtime.Channels do
     query = from c in Channel, where: c.id == ^id
     Repo.del(conn, query)
   end
+
+  @doc """
+  Updates a channel by id from the tenant database using a given DBConnection
+  """
+  @spec update_channel_by_id(binary(), map(), DBConnection.conn()) ::
+          {:ok, Channel.t()} | {:error, any()}
+  def update_channel_by_id(id, attrs, conn) do
+    with {:ok, channel} when not is_nil(channel) <- get_channel_by_id(id, conn),
+         channel <- Channel.changeset(channel, attrs) do
+      Repo.update(conn, channel, Channel)
+    else
+      {:ok, nil} -> {:error, :not_found}
+      error -> error
+    end
```

Comment:
if `get_channel_by_id(id, conn)` instead of `{:ok, nil}` already returns `{:error, :not_found}` then this function and all the other callers can most likely be simpler.

nitpick: `channel <- ` can be pushed out of the `with` given we're not doing a pattern match.

Putting these two ideas together we can end up with the following:

```suggestion
    with {:ok, channel} <- get_channel_by_id(id, conn) do
      changeset = Channel.changeset(channel, attrs)
      Repo.update(conn, channel, Channel)
    end
```

WDYT?